File: solutions/c/darts/1/darts.c
#include "darts.h"

uint8_t score(coordinate_t p)
{
    float dist;
    dist = p.x * p.x + p.y * p.y;
    if (dist > 100) {
        return 0;
    } else if (dist > 25) {
        return 1;
    } else if (dist > 1) {
        return 5;
    }
    return 10;
}

File: solutions/java/darts/1/src/main/java/Darts.java
final class Darts {
    public int score(final double xOfDart, final double yOfDart) {
        final double distanceFromCenter = Math.sqrt(xOfDart * xOfDart + yOfDart * yOfDart);
        if (distanceFromCenter <= 1) {
            return 10;
        } else if (distanceFromCenter <= 5) {
            return 5;
        } else if (distanceFromCenter <= 10) {
            return 1;
        } else {
            return 0;
        }
    }
}
